File: 0241_Different Ways to Add Parentheses/DifferentWaystoAddParentheses.java
class Solution {
    public List<Integer> diffWaysToCompute(String input) {
        List<Integer> ret = new LinkedList<Integer>();
        if(input == null) return ret;
        for(int i=0;i<input.length();i++){
            if(input.charAt(i) == '+' || input.charAt(i) == '-' || input.charAt(i) == '*'){
                List<Integer> left = diffWaysToCompute(input.substring(0,i));
                List<Integer> right = diffWaysToCompute(input.substring(i+1));
                
                for(Integer p1 : left){
                    for(Integer p2 : right){
                        switch(input.charAt(i)){
                            case '+':
                                ret.add(p1 + p2);
                                break;
                            case '-':
                                ret.add(p1 - p2);
                                break;
                            case '*':
                                ret.add(p1 * p2);
                                break;
                        }
                    }
                }
            }
        }
        
        if(ret.size() == 0){
            ret.add(Integer.valueOf(input));
        }
        
        return ret;
    }
}

File: 0241_Different Ways to Add Parentheses/DifferentWaystoAddParentheses.cs
public class Solution {
    public IList<int> DiffWaysToCompute (string input) {
        IList<int> ret = new List<int> ();
        if (string.IsNullOrEmpty (input)) return ret;
        for (int i = 0; i < input.Length; i++) {
            var c = input[i];
            if (c == '+' || c == '-' || c == '*') {
                var left = DiffWaysToCompute (input.Substring (0, i));
                var right = DiffWaysToCompute (input.Substring (i + 1));
                foreach (var p in left) {
                    foreach (var q in right) {
                        switch (c) {
                            case '+':
                                ret.Add (p + q);
                                break;
                            case '-':
                                ret.Add (p - q);
                                break;
                            case '*':
                                ret.Add (p * q);
                                break;
                        }
                    }
                }
            }
        }

        if (ret.Count == 0) ret.Add (System.Convert.ToInt32 (input));
        return ret;
    }
}